mesonbuild/dependencies/mpi.py

```diff
@@ -54,23 +57,45 @@ def mpi_factory(env: 'Environment', for_machine: 'MachineChoice',
     if DependencyMethods.CONFIG_TOOL in methods:
         nwargs = kwargs.copy()
 
-        if compiler_is_intel:
-            if env.machines[for_machine].is_windows():
+        if env.machines[for_machine].is_windows():
+            if compiler_is_intel:
                 nwargs['version_arg'] = '-v'
                 nwargs['returncode_value'] = 3
 
+                if language == 'c':
+                    tool_names = [os.environ.get('I_MPI_CC'), 'mpiicc']
+                elif language == 'cpp':
+                    tool_names = [os.environ.get('I_MPI_CXX'), 'mpiicpc']
+                elif language == 'fortran':
+                    tool_names = [os.environ.get('I_MPI_F90'), 'mpiifort']
+
+                cls_tools = [(IntelMPIConfigToolDependency, tool_names)]  # type: T.List[T.Tuple[T.Type[ConfigToolDependency], T.List[str]]]
+            else:
+                # OpenMPI is not available for Windows
+                # try MS-MPI, including for MSYS2
+                if language == 'c':
+                    tool_names = ['mpicc']
+                elif language == 'cpp':
+                    tool_names = ['mpic++', 'mpicxx']
+                elif language == 'fortran':
+                    tool_names = ['mpifort', 'mpif90', 'mpif77']
+
+                cls_tools = [(MSMPIConfigToolDependency, tool_names)]
+        else:
+            # Linux, MacOS: try the environment variables for the tools first;
+            # fall back to hardcoded names
+
+            # first, IntelMPI
+            # https://software.intel.com/content/www/us/en/develop/documentation/mpi-developer-reference-linux/top/command-reference/compilation-commands.html
             if language == 'c':
-                tool_names = [os.environ.get('I_MPI_CC'), 'mpiicc']
+                tool_names = [os.environ.get('I_MPI_CC'), 'mpigcc']
```

Comment:
I don't have an `mpigcc` installed from openmpi or from intel?

Reply:
mpigcc is from IntelMPI as per the link in the source code just above. It's present on Linux.